klasses/api.py

```diff
@@ -210,3 +212,144 @@ def fetch_bootcamp_run(run_property):
+        (BootcampRunEnrollment): enrollment object that were successfully created
+
+    """
+    enrollment, _ = BootcampRunEnrollment.objects.update_or_create(
+        user=user,
+        bootcamp_run=run,
+        defaults={"active": True, "change_status": None},
+    )
+
+    try:
+        enrollment.user.profile.can_skip_application_steps = True
+        enrollment.user.profile.save()
+    except ObjectDoesNotExist:
+        pass
+
+    if order:  # enrollment created and order is available
+        application = order.application
+        application.bootcamp_run = run
+        application.save()
+
+    if (
+        features.is_enabled(features.NOVOED_INTEGRATION)
+        and enrollment.bootcamp_run.novoed_course_stub
+    ):
+        novoed_tasks.enroll_users_in_novoed_course.delay(
+            user_ids=[enrollment.user.id],
+            novoed_course_stub=enrollment.bootcamp_run.novoed_course_stub,
+        )
+    return enrollment
+
+
+def create_run_enrollments(user, runs, order=None):
+    """
+    Creates local records of a user's enrollment in bootcamp runs, and attempts to enroll them
+    in novoed via API
+
+    Args:
+        user (User): The user to enroll
+        runs (iterable of BootcampRun): The bootcamp runs to enroll in
+        order (ecommerce.models.Order or None): The order associated with these enrollments
+
+    Returns:
+        (list of BootcampRunEnrollment): A list of enrollment objects that were successfully
+            created
+    """
+    successful_enrollments = []
+
+    for run in runs:
+        try:
+            successful_enrollments.append(create_run_enrollment(user, run, order))
+        except:  # pylint: disable=bare-except
+            log.exception(
+                "Failed to create/update enrollment record (user: %s, run: %s, order: %s)",
+                user,
+                run.bootcamp_run_id,
+                order.id if order else None,
+            )
+    return successful_enrollments
+
+
+def defer_enrollment(
+    user, from_bootcamp_run_id, to_bootcamp_run_id, order_id, force=False
+):
+    """
+    Deactivates a user's existing enrollment in one bootcamp run and enrolls the user in another.
+
+    Args:
+        user (User): The enrolled user
+        from_bootcamp_run_id (str): The bootcamp_run_id value of the currently enrolled BootcampRun
+        to_bootcamp_run_id (str): The bootcamp_run_id value of the desired BootcampRun
+        order_id (int): The order_id value for an user's order ID
+        force (bool): If True, the deferral will be completed even if the current enrollment is inactive
+            or the desired enrollment is in a different bootcamp
+
+    Returns:
+        (BootcampRunEnrollment, BootcampRunEnrollment): The deactivated enrollment paired with the
+            new enrollment that was the target of the deferral
+    """
+    from_enrollment = BootcampRunEnrollment.objects.get(
+        user=user, bootcamp_run__bootcamp_run_id=from_bootcamp_run_id
```

Comment:
`BootcampRun.bootcamp_run_id` is a nullable field. I don't think we can use it for this purpose. We should pass in two `BootcampRun` objects for the 'from' and 'to' bootcamp runs, and use a different function for fetching those runs in the management command. If we want to make it user-friendly, we can allow users to specify any one of multiple fields of the `BootcampRun`s they want, like `id`, `title`, or `bootcamp_run_id`

Reply:
@pdpinch what's your opinion on that, should we use `bootcamp_run_id` or we should also add support for (id, title)?

Reply:
Thanks for the discussion. `bootcamp_run_id` is nullable because we have legacy (and test) bootcamp runs that may never get an appropriate `bootcamp_run_id`.

However, any bootcamp run that we can actually defer from or to, **_will_** have a bootcamp_run_id. As an admin, if the field is null, that means I haven't done my job. 

We don't need the added complexity of supporting `id` or `title` inputs.

Reply:
While considering Peter's comment, that's ready for review again.